api/v1alpha1/search_types.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SearchSpec defines the desired state of Search
+type SearchSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// +optional
+	// Storgeclass Name to be used by default db
+	DBStorage StorageSpec `json:"dbStorage,omitempty"`
+
+	// +optional
+	//Configmap name contains parameters to override default db parameters
+	DbConfig string `json:"dbConfig,omitempty"`
+
+	// +optional
+	// Customization for search deployments
+	Deployments SearchDeployments `json:"deployments,omitempty"`
+
+	// +optional
+	// Specifies deployment replication for improved availability. Options are: Basic and High (default)
+	AvailabilityConfig AvailabilityType `json:"availabilityConfig,omitempty"`
+
+	// +optional
+	// Control list of Kubernetes resources indexed by search-collector
+	AllowDenyResources FilterSpec `json:"allowDenyResources,omitempty"`
+
+	// +optional
+	// Kubernetes secret name containing user provided db secret
+	// Secret contains connection url, certificates
+	CustomDbConfig string `json:"customDbConfig,omitempty"`
```

Comment:
Maybe the name could be `ExternalDBInstance`.
What's the expected format for the secret?
Typically certs are managed by a cert service (ie. certmanager) so adding additional fields might be difficult.

Reply:
This secret can like following 
```
data:
        db_url:
        db_name: 
        db_user:
        db_password:
        ca_cert:
```
For example the db can be in a VM , the certs will be provided by the customer right. @jlpadilla

Reply:
- If postgres is on a VM, yes, but most likely there will be some automation to move the certs if these need to be rotated periodically.
- I'm thinking of the scenario where postgres is on the cluster, for example, ACM installs a postgres instance for the product and search uses the ACM postgres instead of our own.

Users will need to know the structure of this secret. Can we add something like this?
```
// ... the secret must contain these keys [db_host, db_port, db_user, db_password, db_name, ca_cert]
```

`db_url` is a compressed format like: `postgresql://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}`

Reply:
This is mainly for ExternalDB - If ACM has a common Postgres instance
(1)  we will update our code to use that secret .